src/doc/src/reference/config.md

```diff
@@ -191,13 +191,14 @@ metadata_key1 = "value"
 metadata_key2 = "value"
 
 [term]
-quiet = false          # whether cargo output is quiet
-verbose = false        # whether cargo provides verbose output
-color = 'auto'         # whether cargo colorizes output
-hyperlinks = true      # whether cargo inserts links into output
-unicode = true         # whether cargo can render output using non-ASCII unicode characters
-progress.when = 'auto' # whether cargo shows progress bar
-progress.width = 80    # width of progress bar
+quiet = false           # whether cargo output is quiet
+verbose = false         # whether cargo provides verbose output
+color = 'auto'          # whether cargo colorizes output
+hyperlinks = true       # whether cargo inserts links into output
+unicode = true          # whether cargo can render output using non-ASCII unicode characters
+progress.when = 'auto'  # whether cargo shows progress bar
+progress.width = 80     # width of progress bar
+progress.taskbar = true # whether cargo reports progress to terminal emulator
```

Comment:
Different terminal emulators handle this differently; for instance, some of the Linux terminal emulators show this in the terminal's tab bar, instead. Rather than having a name that's specific to the rendering of a subset of terminals, could we use a name that's a little more generic?

Perhaps `progress.ansi` or `progress.osc`? Or, if we want something less technical, then perhaps `progress.escapes`, indicating that it sends escape codes?

Reply:
Here are my points:
- `progress.taskbar` - I like this one for easy interchangeability with `taskbar_progress`-named fields and functions
- `progress.terminal` - clearly indicates a connection with terminal emulators
- `progress.osc` - people name PRs and issues mostly with something like "OSC 9;4" so it's somewhat familiar
- `progress.state` - because report is a percentage and state

Reply:
https://github.com/rust-lang/cargo/pull/8165

Reply:
For me, my care abouts are
- Easy discover for people wanting to change this behavior
- Clear meaning when looking at a config or config docs
- Consistency with wider community

I feel like `ansi`, `osc`, `escapes`,  and `osc9_4 do neither unless you are someone already in the nitty gritty implementation details.  Also, our other OSC configs do not speak this way (`color`, `unicode`, `hyperlinks, though `color can also support wincon APIs).

Reply:
> progress.taskbar - I like this one for easy interchangeability with taskbar_progress-named fields and functions

That's circular. :)

Reply:
@epage :+1: to those premises. In addition to those, I'd like to avoid ending up with a name that assumes specifics of how this gets displayed. (This is somewhat similar to https://www.w3.org/QA/Tips/noClickHere .)

Some more possibilities:

- `term.progress.notification` (suggested by @Gordon01)
- `term.progress.terminal` or `term.progress.term` (slightly odd to repeat but they do have distinct meanings)
- `term.progress.terminal-ui`
- `term.progress.ui`
- `term.progress.gui`
- `term.progress.terminal-reporting`

Happy to see any other possibility that doesn't specifically assume a UI display mechanism (e.g. not "title" or "taskbar" or "tab-title" or similar).

Reply:
I don't think there is a clear name to explain itself. `notification` sounds too far away. `osc9_4` is  too tied to implementation details. Maybe we'll have other reporting mechanism for some specific terminal emulator in the future (I hope not though).

Since what the feature does is reporting the progress. So, perhaps a verb `progress.report` is a bit clearer?

(Hey report is also a noun 😅)

Reply:
`notification` can also sound like the kitty feature that slightly overlaps om the codes.

Another name I thought of was `progress.system` to convey that its reporting through the system that calls into cargo.

Eventually I want to add support for multi-line progress which might need its own config and then if we ever get a watch, I would like it to look like `bacon`.  `ui` and `gui` might get confusing in those respects (well, more `tui`).